src/Elcodi/Component/Configuration/Services/ConfigurationManager.php

```diff
@@ -186,13 +187,25 @@ public function get($configurationIdentifier)
 
         if (!($configurationLoaded instanceof ConfigurationInterface)) {
 
-            $parameterReference = $this->configurationElements[$configurationIdentifier]['reference'];
+            $configurationElement = $this->configurationElements[$configurationIdentifier];
+            $configurationValue = isset($configurationElement['reference'])
+                ? $this->parameterBag->get($configurationElement['reference'])
+                : $configurationElement['default_value'];
+
+            if (empty($configurationValue) && !$configurationElement['can_be_empty']) {
```

Comment:
I'd better check for `""` and `null` then.

The magic behind `empty` is scary.

![magic](http://www.reactiongifs.com/r/mgc.gif)